Tools/SwiftBrowser/Source/BrowserViewModel.swift

```diff
@@ -39,17 +39,87 @@ extension PDF: Transferable {
     }
 }
 
+@MainActor
+final class DialogPresenter: DialogPresenting {
+    struct Dialog: Hashable, Identifiable, Sendable {
+        enum Configuration: Sendable {
+            case alert(String, @Sendable () -> Void)
+            case confirm(String, @Sendable (WebPage_v0.JavaScriptConfirmResult) -> Void)
+            case prompt(String, defaultText: String?, @Sendable (WebPage_v0.JavaScriptPromptResult) -> Void)
+        }
+
+        let id = UUID()
+        let configuration: Configuration
+
+        func hash(into hasher: inout Hasher) {
+            self.id.hash(into: &hasher)
```

Comment:
Almost missed this, but what you actually want is
```suggestion
           hasher.combine(id)
```